corehq/apps/hqwebapp/middleware.py

```diff
@@ -27,11 +24,10 @@ def _reject(self, request, reason):
         _assert = soft_assert('{}@{}'.format('sreddy+logs', 'dimagi.com'), exponential_backoff=False)
         _assert(False, warning)
 
-        if '/login/' in request_url:
-            # exempt login URLs from csrf, but still get failure logs
+        if ('/login/' in request_url and
+           settings.CSRF_SOFT_MODE and
+           reason in [REASON_NO_CSRF_COOKIE, REASON_BAD_TOKEN]):
+            # exempt login URLs from csrf if CSRF_SOFT_MODE is on
             self._accept(request)
```

Comment:
`_accept` returns None, `_reject` returns a failure view. Here the rejection code is also run, so it actually rejects. I have also tested this on Prod now. New PR https://github.com/dimagi/commcare-hq/pull/11614